### PR TITLE
Add local profile server devinfo test

### PR DESCRIPTION
**Pull request by eriknordmark**
Tests for shutdown and shutdown_poweroff in the Local Profile Server DevInfo API while checking that an app
instance running a local profile server is shut down last.

Separate commit for a test for shutdown from controller.

Depends on the implementation in https://github.com/lf-edge/eve/pull/2610

### CODE REVIEW

File: tests/eclient/testdata/dev_local_info.txt
Comment (giggsoff) [inline tests/eclient/testdata/dev_local_info.txt:32]: Version of eclient should be updated after image push, and we push images on PR merge right now. So please split PR onto two: one for update of eclient image and the second one for update tests.

File: tests/eclient/testdata/dev_local_info.txt
Comment (giggsoff) [inline tests/eclient/testdata/dev_local_info.txt:49]: I think it will be nice to add sleep for i.e. 30 seconds before this step.

File: tests/eclient/testdata/dev_local_info.txt
Comment (giggsoff) [inline tests/eclient/testdata/dev_local_info.txt:118]: It should be `eden eve start`. Reset command will remove all apps from EVE configuration but will not affect qemu process state. In general seems we can run this only for qemu (see another comment).

File: tests/eclient/testdata/dev_local_info.txt
Comment (giggsoff) [inline tests/eclient/testdata/dev_local_info.txt:142]: We should put `eden eve reset` and small sleep here to remove configuration of LPS from EVE config.

File: tests/eclient/testdata/dev_local_info.txt
Comment (giggsoff) [inline tests/eclient/testdata/dev_local_info.txt:219]: As I understand it should be json representation of [LocalDevCmd](https://github.com/lf-edge/eve/blob/master/api/go/profile/local_profile.pb.go#L815-L826), so command is enough. Also I can see only one argument for calling this function, so please remove DN and TIMESTAMP.

File: tests/eclient/testdata/dev_local_info.txt
Comment (giggsoff) [inline tests/eclient/testdata/dev_local_info.txt:93]: We can run the rest of the test only with qemu right now to properly handle start of EVE after poweroff.
Please add the following before the step 6:
```
{{$devmodel := EdenConfig "eve.devmodel"}}
{{if not (eq $devmodel "ZedVirtual-4G")}}
eden eve reset
skip 'The rest of the test is supported only on QEMU'
{{end}}
```

File: tests/eclient/testdata/dev_local_info.txt
Comment (giggsoff) [inline tests/eclient/testdata/dev_local_info.txt:21]: It should catch 3 reboots to fail (2 reboots are expected, the third one is an error).

**Review by giggsoff — CHANGES_REQUESTED** [review]
Please put eclient image update into the separate PR and see comments

File: tests/eclient/testdata/dev_local_info.txt
Comment (eriknordmark) [reply tests/eclient/testdata/dev_local_info.txt:32]: Ah - that docker tag is what selects the version. Got it.

Comment (eriknordmark) [reply tests/eclient/testdata/dev_local_info.txt:32]: Created a separate https://github.com/lf-edge/eden/pull/762 with eclient.

File: tests/eclient/testdata/dev_local_info.txt
Comment (eriknordmark) [reply tests/eclient/testdata/dev_local_info.txt:49]: OK

File: tests/eclient/testdata/dev_local_info.txt
Comment (eriknordmark) [reply tests/eclient/testdata/dev_local_info.txt:118]: OK

File: tests/eclient/testdata/dev_local_info.txt
Comment (eriknordmark) [reply tests/eclient/testdata/dev_local_info.txt:142]: Before the pod delete? Or before the delete of the local-manager?

I don't think we have that step in the other uses of the local profile server; I copied this from app_local_info.txt

File: tests/eclient/testdata/dev_local_info.txt
Comment (eriknordmark) [reply tests/eclient/testdata/dev_local_info.txt:93]: Thanks

File: tests/eclient/testdata/dev_local_info.txt
Comment (eriknordmark) [reply tests/eclient/testdata/dev_local_info.txt:21]: OK, didn't realize the limit had that semantics; thought it was the expected number not the number at which it is a failure.

File: tests/eclient/testdata/dev_local_info.txt
Comment (giggsoff) [reply tests/eclient/testdata/dev_local_info.txt]: Timestamp set automatically based on file modification time here: https://github.com/lf-edge/eden/blob/99d7f6244a1b98b3fb1b3372f0346bb44babebe1/tests/eclient/image/pkg/main.go#L219

File: tests/eclient/testdata/dev_local_info.txt
Comment (giggsoff) [reply tests/eclient/testdata/dev_local_info.txt:151]: Oh, seems I missed line here, after waiting for network delete, in the end.
Also please update eve version to one with PR merged here https://github.com/lf-edge/eden/blob/99d7f6244a1b98b3fb1b3372f0346bb44babebe1/pkg/defaults/defaults.go#L53

File: tests/eclient/testdata/dev_local_info.txt
Comment (eriknordmark) [reply tests/eclient/testdata/dev_local_info.txt]: Right. Fixing that.

File: tests/eclient/testdata/dev_local_info.txt
Comment (eriknordmark) [reply tests/eclient/testdata/dev_local_info.txt:151]: When do we update DefaultEVETag? Any time there is a new feature in EVE-OS needed to pass the tests?

Comment (giggsoff) [reply tests/eclient/testdata/dev_local_info.txt:151]: Yes, we update version inside Eden repo when we have new feature in EVE test need to run. Inside EVE repo we use current (the last) version of EVE for tests.

File: tests/eclient/testdata/dev_local_info.txt
Comment (eriknordmark) [reply tests/eclient/testdata/dev_local_info.txt:97]: But we can't wait for the devinfo as reported by the local-manager since it isn't running. Do we have some function to look at the relatively new state reported by the device itself in ZDeviceState?

One thing I need to check is how EVE-OS decides to back-off the LPS API calls. It might need some patience after a device reboot since the LPS might not be running yet.

Comment (eriknordmark) [reply tests/eclient/testdata/dev_local_info.txt:97]: @giggsoff I don't see any match on ZDeviceState in eden source code.

Comment (giggsoff) [reply tests/eclient/testdata/dev_local_info.txt:97]: I mean you can call script `wait-for-dev-state.sh` with ONLINE match instead of `get-devinfo-status.sh`. With this change we will use loop and will check the file inside LPS several times.

File: tests/eclient/testdata/shutdown_test.txt
Comment (giggsoff) [inline tests/eclient/testdata/shutdown_test.txt:34]: Please sync timewaits inside this block. Timewait for app and network should be more than timewait for reboot.

File: tests/eclient/testdata/dev_local_info.txt
Comment (eriknordmark) [reply tests/eclient/testdata/dev_local_info.txt:97]: Sure I can change that (I thought this check was before the LPS was started but it is after it is running.)

File: tests/eclient/testdata/shutdown_test.txt
Comment (eriknordmark) [reply tests/eclient/testdata/shutdown_test.txt:34]: Updated

**Review by giggsoff — APPROVED** [review]
LGTM

will work on SHUTTING_DOWN state check in next PR

**Pull request merged**
